src/foam/nanos/dig/DUGRuleAction.js

```diff
@@ -45,68 +49,121 @@ foam.CLASS({
-          DUGDigestConfig dugDigestConfig = (DUGDigestConfig) dugDigestConfigDAO.find(rule.getSpid());
-          try {
-            AbstractSink sink = null;
-            if ( dugDigestConfig != null && dugDigestConfig.getEnabled() ) {
-              sink = new HTTPDigestSink(
-                dugRule.getUrl(),
-                dugRule.evaluateBearerToken(),
-                dugDigestConfig,
-                dugRule.getFormat(),
-                new foam.lib.AndPropertyPredicate(
-                  x,
-                  new foam.lib.PropertyPredicate[] {
-                    new foam.lib.ExternalPropertyPredicate(),
-                    new foam.lib.NetworkPropertyPredicate(),
-                    new foam.lib.PermissionedPropertyPredicate()
-                  }
-                ),
-                true,
-                true
-              );
-            } else {
-              sink = new HTTPSink(
-                dugRule.getUrl(),
-                dugRule.evaluateBearerToken(),
-                dugRule.getFormat(),
-                new foam.lib.AndPropertyPredicate(
-                  x,
-                  new foam.lib.PropertyPredicate[] {
-                    new foam.lib.ExternalPropertyPredicate(),
-                    new foam.lib.NetworkPropertyPredicate(),
-                    new foam.lib.PermissionedPropertyPredicate()
-                  }
-                ),
-                true
-              );
-            }
-
-            sink.setX(x);
-            sink.put(obj, null);
-          } catch (Throwable t) {
-            ((Logger) x.get("logger")).error(this.getClass().getSimpleName(), "Error Sending DUG webhook", t);
-            var alarmDAO = (DAO) x.get("alarmDAO");
-            alarmDAO.put(
-              new Alarm.Builder(x)
-                .setName("DUG/"+rule.getDaoKey() + "/" + rule.getName())
-                .build()
+
+      // If the actingUser is set, we want to refind the obj with that user as the subject
+      // The goal with this is to filter the permissioned properties based on acting user
+      // Instead of the user that triggered the rule
+
+      final var actingUserId = dugRule.getActingUser();
+
+      if ( actingUserId != 0 ) {
+        final var userDAO = (DAO) x.get("bareUserDAO");
+        final var user = (User) userDAO.find(actingUserId);
+
+        if ( user == null ) {
+          getLogger(x).error("DUGRule ", dugRule.getId(), " has acting user but user couldn't be found");
+          return;
+        }
+
+        final var groupDAO = (DAO) x.get("localGroupDAO");
+        final var group = groupDAO.find(user.getGroup());
+
+        if ( group == null ) {
+          getLogger(x).error("DUGRule ", dugRule.getId(), " has acting user but group couldn't be found");
+          return;
+        }
+
+        if ( SafetyUtil.isEmpty(dugRule.getSecureDaoKey()) ) {
+          getLogger(x).error("DUGRule ", dugRule.getId(), " has acting user but secure DAO key couldn't be found");
+          return;
```

Comment:
instead of failing here .. you should just use the daoKey .. looks like you are already doing that below .. i think you can just remove the return statement.

Reply:
Yeah that was my intention, forgot the remove the return, thanks